open_in_mpv/open_in_mpv.py

```diff
@@ -15,32 +14,90 @@
 import click
 import xdg.BaseDirectory
 
+from .constants import IS_MAC, IS_WIN, MACPORTS_BIN_PATH
+
+FALLBACKS: Final[dict[str, Any]] = {
+    'log': None,
+    'socket': None
+}
+
 
 @lru_cache()
 def get_log_path() -> str:
-    if platform.mac_ver()[0]:
+    if IS_MAC:
         return expanduser('~/Library/Logs')
+    if IS_WIN:
+        return expandvars(r'%LOCALDATA%\open-in-mpv') # cspell:disable-line
     try:
         return xdg.BaseDirectory.save_state_path('open-in-mpv')
     except KeyError:
-        with tempfile.TemporaryDirectory(prefix='open-in-mpv') as dir_:
-            return dir_
+        FALLBACKS['log'] = tempfile.TemporaryDirectory(prefix='open-in-mpv') # pylint: disable=R1732
+        return str(FALLBACKS['log'].name)
 
 
 @lru_cache()
 def get_socket_path() -> str:
-    if platform.mac_ver()[0]:
+    if IS_MAC:
         return expanduser('~/Library/Caches/open-in-mpv.sock')
+    if IS_WIN:
+        return expandvars(r'\\.\pipe\open-in-mpv')
     try:
         return path_join(xdg.BaseDirectory.get_runtime_dir(), 'open-in-mpv.sock')
     except KeyError:
-        with tempfile.NamedTemporaryFile(prefix='open-in-mpv', suffix='.sock') as socket_fp:
-            return socket_fp.name
-
+        FALLBACKS['socket'] = tempfile.NamedTemporaryFile(prefix='open-in-mpv', suffix='.sock') # pylint: disable=R1732
+        return str(FALLBACKS['socket'].name)
 
 LOG_PATH = get_log_path()
 MPV_SOCKET = get_socket_path()
-VERSION = 'v0.0.6'
+VERSION = 'v0.1.7'
+
+
+def environment(data_resp: dict[str, Any], debugging: bool) -> dict[str, Any]:
+    env: dict[str, Any] = os.environ.copy()
+    if isdir(MACPORTS_BIN_PATH):
+        logger.info('Detected MacPorts. Setting PATH.')
+        data_resp['macports'] = True
+        old_path = os.environ.get('PATH')
+        env['PATH'] = MACPORTS_BIN_PATH if not old_path else ':'.join(MACPORTS_BIN_PATH, old_path)
```

Comment:
This `join` call is incorrect. Put the arguments in a tuple: `':'.join((MACPORTS_BIN_PATH, old_path))`